src/components/ShopPage/ShopPageHero/index.tsx

```diff
@@ -0,0 +1,117 @@
+        toaster.notify(props.shop.phone)
+    }
+
+    const cityState = (): string => {
+        if (props.shop.city && props.shop.state){
+            return `${props.shop.city}, ${props.shop.state}`
+        }
+        else if (props.shop.city && props.shop.country){
+            return `${props.shop.city}, ${props.shop.country_name}`
+        }
+        else if (props.shop.city){
+            return `${props.shop.city}`
+        }
+        else if (props.shop.state){
+            return props.shop.state
+        }
+        else if (props.shop.country){
+            return props.shop.country_name
+        }
+        else {
+            return ""
+        }
+    }
+
+    return (
+        <div className={styles.image}>
+            <div className={styles.imageinner}>
+                {props.shop.logo_img && <Image
+                    src='/hero.jpg'
+                    layout='fill'
+                    objectFit='cover'
+                    alt={props.shop.name}
+                />}
+                <div className={styles.overlay} />
+                <div className={styles.pageHeroInfo}>
+                    <div className={styles.nameContainer}>
+                        <h1 className={styles.pagetitle}>{props.shop.name}</h1>
+                    </div>
+                    <div className={styles.stars}>
+                        <Rating
+                            fractions={2}
+                            emptySymbol={(<EmptyStar />)}
+                            fullSymbol={(<FullStar />)}
+                            initialRating={props.shop.rating}
+                            readonly>
+                        </Rating>
+                    </div>
+                    <div>
+                    {cityState()}
+                    </div>
+                </div>
+            </div>
+            <div className={styles.menu}>
+                <div className={styles.buttoncontainer}>
+                    <div className={styles.buttonouter}>
+                        <a className={styles.buttoncircle} onClick={() => contactClick()}>
+
+                            <Image src='/contactphone.png' alt="phone" objectFit="contain" height='20' width="20"></Image>
+                        </a>
+                        <div className={styles.buttonlabel}>Contact</div>
+                    </div>
+                    <div className={styles.buttonouter}>
+                            <a className={styles.buttoncircle} href={props.shop.url}>
+                                <Image src='/iconoir_internet.png' alt="map" objectFit="contain" height="20" width="20"></Image>
+                            </a>
+                        <div className={styles.buttonlabel}>Website</div>
+                    </div>
+                    <div className={styles.buttonouter}>
+                        {
+                            props.shop.location_google ? (
+                                <Link href={props.shop.location_google || ''}>
```

Comment:
for the link here we need to create a google maps link that looks up the address. i think just using the query format here: https://developers.google.com/maps/documentation/urls/get-started